### PR TITLE
Feat: moderation app moderators page

### PR DESCRIPTION
**Pull request by josenriagu**
## Description

- update moderation app moderators page

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed
- [x] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/design-system-core/src/components/BasicCardBox/index.tsx [hunk 1]
```diff
@@ -45,10 +47,10 @@ const BasicCardBox: React.FC<PropsWithChildren<IBasicCardBox>> = props => {
 
   const className = `flex flex-col shadow-${elevation} w-full ${pad ? pad : 'p-0'} ${
```
Comment (quininez) [inline ui/design-system-core/src/components/BasicCardBox/index.tsx]: here you can use the cx utility from twind: const class = cx(`flex flex-col....) and on line 53 just use `<div` className={class}>`

File: ui/design-system-core/src/components/ModerationSwitchCard/index.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+    <div className={tw('flex w-full')}>
+      {tabs.map((tab, idx) => (
+        <div
+          key={tab.title + idx}
+          className={tw(
+            `w-[25%] p-4 border-b ${
+              tab.value === activeTab
+                ? ' border-secondary-light dark:border-secondary-dark'
+                : 'border-grey8 dark:border-grey3'
```
Comment (quininez) [inline ui/design-system-core/src/components/ModerationSwitchCard/index.tsx]: you can use the [grouping syntax](https://twind.style/grouping-syntax) like border(grey8 dark:grey3)

File: ui/design-system-core/src/components/ModeratorDetailCard/index.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+  };
+
+  const truncateStyle =
+    'max-w-[12.5rem] md:max-w-[7.5rem] w-fit whitespace-nowrap overflow-hidden text-ellipsis cursor-default';
+
+  return (
+    <div
+      className={tw(
+        `flex py-4 flex-none ${
+          hasBorderBottom ? 'border-b-1 border-solid border-grey8 dark:border-grey3' : ''
```
Comment (quininez) [inline ui/design-system-core/src/components/ModeratorDetailCard/index.tsx]: use grouping syntax to make it more readable border(b-1 solid grey8 dark:grey-3)

File: ui/design-system-core/src/components/ModeratorDetailCard/index.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+        }`,
+      )}
+    >
+      <div
+        className={tw(
+          'flex space-x-2 items-start w-[50%] md:w-[30%] px-4 border-r-1 border-solid border-grey8 dark:border-grey3',
```
Comment (quininez) [inline ui/design-system-core/src/components/ModeratorDetailCard/index.tsx]: grouping

File: ui/design-system-core/src/components/ModeratorDetailCard/index.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+              variant="button-md"
+              weight="normal"
+              className={tw(truncateStyle)}
+              color={{ light: 'text-grey4', dark: 'text-grey7' }}
+            >{`@${moderator.userName}`}</Text>
+          </Tooltip>
+          <div className={tw('flex space-x-1.5 items-center')}>
+            <div
+              className={tw(
+                `w-1.5 h-1.5 rounded-full ${
```
Comment (quininez) [inline ui/design-system-core/src/components/ModeratorDetailCard/index.tsx]: refactor this, extract conditionals in outside variables

File: ui/design-system-core/src/components/ModeratorDetailCard/social-link.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  iconSize?: IAppIcon['size'];
+  onClick: () => void;
+}
+
+const SocialLink: React.FC<ISocialLinksProps> = props => {
+  const { iconType, iconSize = 'md', onClick } = props;
+
+  return (
+    <div
+      className={tw(`flex items-center justify-center w-8 h-8 rounded-full bg-grey9 dark:bg-grey3`)}
```
Comment (quininez) [inline ui/design-system-core/src/components/ModeratorDetailCard/social-link.tsx]: grouping wherever its possible bg(grey9 dark:grey3)

File: ui/design-system-core/src/components/Spinner/index.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+const Spinner: React.FC<ISpinnerProps> = props => {
+  const { size = 'md', color = 'current' } = props;
+
+  return (
+    <div role="status">
+      <svg
+        aria-hidden="true"
+        className={tw(
+          apply(
```
Comment (quininez) [inline ui/design-system-core/src/components/Spinner/index.tsx]: no need to use apply within tx like this, you can use directly tw or cx utility

**Review by quininez — CHANGES_REQUESTED** [review]

File: ui/apps/moderation/src/pages/moderators.tsx [hunk 8]
```diff
@@ -43,68 +29,46 @@ const AllModerators: React.FC<RootComponentProps> = () => {
...
+          <Spinner size="lg" />
+        </Box>
+      )}
 
       {!getModeratorsQuery.isFetching &&
         getModeratorsQuery.data &&
         getModeratorsQuery.data.length > 0 && (
-          <ListWrapper>
-            <ListArea>
+          <Box style="flex-1">
```
Comment (quininez) [inline ui/apps/moderation/src/pages/moderators.tsx]: I think it's better to wrap the classes with tw or tx utilities just to make sure we don't have issues going further

**Review by quininez — CHANGES_REQUESTED** [review]

File: ui/apps/moderation/src/pages/moderators.tsx (hunk 8 above)
Comment (josenriagu) [reply ui/apps/moderation/src/pages/moderators.tsx]: I changed the style prop on the other PR, so I will still update this when it's merged

**Review by quininez — APPROVED** [review]

**Pull request merged**